docs/design/jet/025-jet-job-partition-pruning.md

```diff
@@ -123,9 +123,85 @@ try to choose only the required members.
+partition pruning candidates as a parameter and calculate it in connector-specific way:
+```
+    @Override
+    @Nonnull
+    public Vertex fullScanReader(
+            @Nonnull DagBuildContext context,
+            @Nullable HazelcastRexNode predicate,
+            @Nonnull List<HazelcastRexNode> projection,
+            @Nullable List<Map<String, Expression<?>>> partitionPruningCandidates, // <-- new parameter
+            @Nullable FunctionEx<ExpressionEvalContext, EventTimePolicy<JetSqlRow>> eventTimePolicyProvider)
+```
+
+The new parameter `partitionPruningCandidates` is a list of maps, where for each column present in the filter 
+the column name maps to the extracted comparison expression. The connector-specific implementation should
+filter predicates for columns where partitioning strategy is applicable, and check if partitioning strategy 
+is applicable generally, and in case of success transform the input to the **list of inner list of expressions**. 
+Each inner list of expressions contains comparison expressions, and may be **one-sized**, 
+if the partitioning strategy key is **simple**, and **multi-sized**, if the partitioning strategy key is **composite**. 
+If we have more than one prunable filter predicate, outer list will be multi-sized. 
+
+**Currently, it is implemented only for IMap connector, where all expressions are available.**
+
+For better imagination we prepared an example below.
+
+#### Successful case example 
+Let's assume we have an IMap `map` with composite key {comp1, comp2, comp3} and applied partitioning strategy for
+`comp1` and `comp2`. Let's have the following synthetic query, where filter matches the partitioning strategy:
+```
+SELECT * FROM map WHERE __key.comp1 = 1 AND __key.comp2 = 2
+```
+IMap-specific `fullScanReader` receives the following list of maps as a parameter:
+```
+[{"__key.comp1" = Expression(`__key.comp1 = 1`)}, {"__key.comp2" = Expression(`__key.comp2 = 2`)}]
+```
+
+After the described computation above, `fullScanReader` implementation should return the following list of expressions:
+```
+[[Expression(`__key.comp1 = 2`], [Expression(`__key.comp1 = 2`]]
+```
+
+#### Not successful case example
+
+For same `map` as above, let's have the following synthetic query, where filter does not match the partitioning strategy:
+```
+SELECT * FROM map WHERE __key.comp1 = 1 AND __key.comp3 = 3
+```
+
+IMap-specific `fullScanReader` receives the following list of maps as a parameter:
+```
+[{"__key.comp1" = Expression(`__key.comp1 = 1`)}, {"__key.comp2" = Expression(`__key.comp3 = 3)}]
+```
+
+but, instead of example below, it will return an empty list, because partition pruning is not applicable due to presence
+of `__key.comp3` in the filter, which was not declared as part of partitioning strategy.
+
+**Important note : simple `__key` is always participates in default partitioning strategy, since IMap is partitioned
+by __key**.
```

Comment:
nope, in case of AttributeParitioningStrategy the IMap is partitioned by _function of key_ not by the key.

Reply:
Its a `DefaultParitioningStrategy`

Reply:
It's gone : https://github.com/hazelcast/hazelcast/issues/25033

Reply:
it will be back: https://hazelcast.atlassian.net/browse/HZ-2525